htlcswitch/link.go

```diff
@@ -1836,15 +1833,22 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		l.cfg.Peer.SendMessage(false, nextRevocation)
 
 		// Since we just revoked our commitment, we may have a new set
-		// of HTLC's on our commitment, so we'll send them over our
-		// HTLC update channel so any callers can be notified.
-		select {
-		case l.htlcUpdates <- &contractcourt.ContractUpdate{
+		// of HTLC's on our commitment, so we'll send them using our
+		// function closure NotifyContractUpdate.
+		newUpdate := &contractcourt.ContractUpdate{
 			HtlcKey: contractcourt.LocalHtlcSet,
 			Htlcs:   currentHtlcs,
-		}:
+		}
+		err = l.cfg.NotifyContractUpdate(newUpdate)
```

Comment:
So the only error that will be returned here is the channel point cannot be found?

Meanwhile, this method is now blocking. Unlike the previous `select` statement, we might be stuck here and cannot quit?

Reply:
what if the `ChannelArbitrator` hasn't been started yet?

Reply:
channelarbitrator are all started when chainarbitrator starts. startup order doesn't matter here - links are only started once connections are initiated which would happen at the end of the server start routine. links aren't started when the switch starts, so this can't happen

Reply:
Cool thanks for clarifying!